### ISSUE TITLE
Media Engine / HEVEC / ProRes Usage

### ISSUE THREAD

**Issue by chrisspiegl** [tlkh/asitop#13]
Hi @tlkh , I just found out about this tool and love how much information it displays very easily.

I was wondering if it would be possible to visualize the usage of the Media Engines in a similar Historgram / Graph way as CPU Usage and such? And is it possible to separate out the ProRes and h265/264 engines?

I am wondering because Apple advertised that the M1Pro and M1Max have different counts of these engines and I would wonder if it's possible to actually see when these engines are in use vs. when they are not (for example, love that with your tool it's possible to see the ANE usage when it is utilized).

TLDR:

Would it be possible to have a usage graph of the individual Media Engines?

Thanks and I am looking forward to updates to this powerful GUI 👍.

**Comment by tlkh** [comment]
Not that I am aware of.

For ANE, the proxy to measuring utilisation is power usage (we cannot check the ANE utilisation directly, but power usage is a good proxy)

For Media Engine, all we have is the bandwidth information. The combined media engine bandwidth is already presented in asitop.  Here is what I can see exists on my M1 Max, and M1 Pro would be missing some:

- ISP (camera only, I think)
- STRM CODEC (not sure exactly what this is for)
- PRORES (no 0/1 !!)
- VDEC (H264/H265, I think)
- VENC0 and VENC1 (2 engines)
- JPG0 and JPG1 (2 engines)

Currently, asitop reports the total bandwidth usage for all these engines. 

I am thinking about how to include in more information, at some point it would not fit well into a single window. I am considering making multiple pages/screen for certain detailed displays of specific components.

**Comment by chrisspiegl** [comment]
Thank you for your detailed reply. That is great information. Too bad Apples powermetrics only provides the bandwidth for the Media Engine. But at least we have some. I would love to see the details output either via pages or maybe via a command on the CLI call of `asitop`.

One can resize the window on bigger screens so providing it with a parameter `--full` or `--detailed-media-engine` could be an option.

Overall, I am excited to see more information about this available in some way (ideally even with historical data / graphs) to fully be able to understand which Media Engine is utilized.

Thanks again and I'm open for alpha/beta testing if you have anything.
